lib/src/edhoc.rs

```diff
@@ -853,20 +879,10 @@ fn encode_plaintext_2(
 ) -> Result<BufferPlaintext2, EDHOCError> {
     let mut plaintext_2: BufferPlaintext2 = BufferPlaintext2::new();
     let c_r = c_r.as_slice();
-    plaintext_2.content[..c_r.len()].copy_from_slice(c_r);
 
-    let offset_cred = match id_cred_r {
-        IdCred::CompactKid(kid) => {
-            plaintext_2.content[c_r.len()] = *kid;
-            c_r.len() + 1
-        }
-        IdCred::FullCredential(cred) => {
-            plaintext_2.content[c_r.len()] = CBOR_BYTE_STRING;
-            plaintext_2.content[c_r.len() + 1] = cred.len() as u8;
-            plaintext_2.content[c_r.len() + 2..][..cred.len()].copy_from_slice(cred);
-            c_r.len() + 2 + cred.len()
-        }
-    };
+    plaintext_2.extend_from_slice(c_r).unwrap();
+    id_cred_r.write(&mut plaintext_2)?;
```

Comment:
Same as above.

Reply:
Renamed in 0f2fef8994262c988bc54690a32f5e1a84f96200